### PR TITLE
Nested end to end pipeline deploy

### PR DESCRIPTION
**Pull request by huguesBouvier**
Deploys IIOT on lvl 4 and 5:
https://dev.azure.com/msazure/One/_build/results?buildId=38618098&view=results

Regression test:
https://dev.azure.com/msazure/One/_build/results?buildId=38618933&view=logs&j=4323b27c-bad1-54aa-e27b-49b281e4d21f

### CODE REVIEW

File: builds/e2e/templates/e2e-run.yaml [hunk 1]
```diff
@@ -13,13 +17,19 @@ steps:
     {
       $filter += '&Category!=UnstableOnArm'
     }
-
-    sudo --preserve-env dotnet test $testFile --logger:trx --testcasefilter:$filter
+    
+    #filter out tests not usable on nested edge
+    if ('$(nestededge)' -eq 'true')
+    {
+      #$filter += 'FullyQualifiedName!~Provisioning&FullyQualifiedName!~PlugAndPlay&FullyQualifiedName!~PriorityQueueModuleToHubMessages&FullyQualifiedName!~SasOutOfScope'
```
Comment (yophilav) [inline builds/e2e/templates/e2e-run.yaml:24]: Please remove unused command.

File: builds/e2e/templates/nested-agent-deploy.yaml [hunk 2]
```diff
@@ -45,19 +49,35 @@ steps:
...
-    displayName: 'Copy Images Artifact'
+      filePath: $(Build.SourcesDirectory)/scripts/local/test/DownloadIdentityService.ps1
+      workingDirectory: $(Build.SourcesDirectory)
+    env:
+      GITHUB_TOKEN: $(GitHubAccessToken)
+      ARTIFACT_NAME: $(identityServiceArtifactName)
+      PACKAGE_FILTER: $(identityServicePackageFilter)
+      DOWNLOAD_PATH: $(Build.StagingDirectory) 
+  - task: Bash@3     
+    displayName: 'Install artifacts'
```
Comment (yophilav) [inline builds/e2e/templates/nested-agent-deploy.yaml:64]: This is actually managed within E2E test framework. THe pipline shouldn't be the one dealing with this step :  https://github.com/Azure/iotedge/blob/master/test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManage
ment.cs

File: builds/e2e/templates/nested-get-secrets.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ steps:
     inputs:
       azureSubscription: $(az.subscription)
       KeyVaultName: $(azure.keyVault)
-      SecretsFilter: 'edgebuilds-azurecr-io-username,edgebuilds-azurecr-io-pwd,kvLogAnalyticWorkspaceId,kvLogAnalyticSharedKey,EdgeConnectivityStorageAccountConnString'
+      SecretsFilter: 'edgebuilds-azurecr-io-username,edgebuilds-azurecr-io-pwd,kvLogAnalyticWorkspaceId,kvLogAnalyticSharedKey,EdgeConnectivityStorageAccountConnString,GitHubAccessToken'
```
Comment (yophilav) [inline builds/e2e/templates/nested-get-secrets.yaml:7]: why do we need `GitHubAccessToken` ?

**Review by yophilav — COMMENTED** [review]
Given the nature of the work & intrinsic different in `Master` and `release 1.2`. 
There are going to be quite a bit of changes done to ensure a successful run on both branch.
Since `release/1.2` is closer to the existing version of E2E maybe we should try to merge the Nested E2E there first to ensure the stability of the E2E pipeline we are creating. Once that's stable, we can port over necessary change in master?

File: builds/e2e/templates/nested-agent-deploy.yaml (hunk 2 above)
Comment (huguesBouvier) [reply builds/e2e/templates/nested-agent-deploy.yaml:64]: The E2E framework doesn't run on lvl4/5

File: builds/e2e/templates/nested-get-secrets.yaml (hunk 3 above)
Comment (huguesBouvier) [reply builds/e2e/templates/nested-get-secrets.yaml:7]: Not used for us. Used y new IIOT to get library

File: test/Microsoft.Azure.Devices.Edge.Test.Common/LeafDevice.cs [hunk 4]
```diff
@@ -127,6 +130,13 @@ public class LeafDevice
                     leaf.Scope = edge.Scope;
                 });
 
+            // @To Remove this is a hack to be able to create lea. See https://dev.azure.com/msazure/One/_workitems/edit/9171870
```
Comment (richma-ms) [inline test/Microsoft.Azure.Devices.Edge.Test.Common/LeafDevice.cs:133]: >https://dev.azure.com/msazure/One/_workitems/edit/9171870 [](start = 71, length = 57)

just reference the PBI number

File: builds/e2e/templates/e2e-run.yaml [hunk 5]
```diff
@@ -13,13 +17,20 @@ steps:
     {
       $filter += '&Category!=UnstableOnArm'
     }
-
-    sudo --preserve-env dotnet test $testFile --logger:trx --testcasefilter:$filter
+    
+    #filter out tests not usable on nested edge
+    if ('$(nestededge)' -eq 'true')
+    {
+      #We are disabling some additional tests. See  https://dev.azure.com/msazure/One/_workitems/edit/9171870. Once this is fixed we should run what's directly below.
```
Comment (richma-ms) [inline builds/e2e/templates/e2e-run.yaml:24]: >https://dev.azure.com/msazure/One/_workitems/edit/9171870 [](start = 52, length = 57)

just reference the PBI number

File: test/Microsoft.Azure.Devices.Edge.Test.Common/LeafDevice.cs (hunk 4 above)
Comment (huguesBouvier) [reply test/Microsoft.Azure.Devices.Edge.Test.Common/LeafDevice.cs]: Fixed

File: builds/e2e/templates/e2e-run.yaml (hunk 5 above)
Comment (huguesBouvier) [reply builds/e2e/templates/e2e-run.yaml]: Fixed

**Review by richma-ms — APPROVED** [review]

**Pull request merged**